### ISSUE TITLE
Intellisense Clang VLA

### ISSUE THREAD

**Issue by JonasSchlatter** [microsoft/vscode-cpptools#6995]
Issue Type: <b>Bug</b>

Using `"intelliSenseMode": "windows-clang-x64"` the declaration of a VLA

```
int size = 2;
double array[size];
```

warns that the expression must have a constant value; Clang however does support VLAs.

The same issue was raised (and resolved) for Visual Studio:
https://developercommunity.visualstudio.com/content/problem/871054/incorrect-intellisense-errors-when-using-clang.html

Extension version: 1.2.2-insiders
VS Code version: Code 1.53.2 (622cb03f7e070a9670c94bae1a45d78d7181fbd4, 2021-02-11T11:48:04.245Z)
OS version: Windows_NT x64 10.0.19042

**Comment by sean-mcmanus** [comment]
It's supposed to work (works for me). Can you run C/C++: Log Diagnostics? It looks like it's switching from windows-clang-x64 to windows-msvc-x64 mode probably.

**Comment by JonasSchlatter** [comment]
Current Configuration:
`    "intelliSenseMode": "windows-clang-x64",
`

Custom Configurations:
`    "intelliSenseMode": "clang-x64",
`

Translation Unit Configurations:
`    IntelliSense Mode: windows-clang-x64
`

Could there be an issue linked to the occurrence of
`--ms_extensions` or `"--dependent-lib=msvcrtd"`?

**Comment by JonasSchlatter** [comment]
Switching between my (working) GCC configuration and the Clang config, probing C/C++: Log Diagnostics there's one difference between the Current Configurations. The Clang config adds:
```
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "compilerPathIsExplicit": true,
```

**Comment by sean-mcmanus** [comment]
Hmm, all that looks okay. Do you see anything that could be significantly different with what I'm getting? Is your code in a function right? Not a global, which shouldn't compile.

![image](https://user-images.githubusercontent.com/19859882/108451222-4a3b0f00-721b-11eb-99ea-35c9d0b001bb.png)

**Comment by JonasSchlatter** [comment]
Ah! The significant difference is the file extension.

If I rename my file to .cpp it works.
If I rename it back to .c it doesn't.

**Comment by sean-mcmanus** [comment]
Ah, okay, it's a duplicate of https://github.com/microsoft/vscode-cpptools/issues/6500 , which is tracking https://developercommunity.visualstudio.com/content/problem/1256096/cc-intellisense-gives-parameter-is-not-allowed-for.html (which you can upvote).

**Comment by JonasSchlatter** [comment]
Thank you for your help, upvoted the issue. Probably closely related. Although #6500's "parameter not allowed" within a formal parameter doesn't work for both C and C++, VLA declaration within a function is broken for .c files only.

C (allows neither)
![cfile](https://user-images.githubusercontent.com/19890027/108455725-08d44f00-726f-11eb-9428-18e68449d06f.png)

C++ (allows VLA declaration)
![cppfile](https://user-images.githubusercontent.com/19890027/108455728-0b36a900-726f-11eb-9d74-2cca56973b61.png)

**Comment by sean-mcmanus** [comment]
Yeah, the C++ parameter list is a different case, but the other cases are the same root issue. Should be fixed with 1.3.0-insiders (unless it makes it into 1.2.2).

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.2.2
